File: AppMovies/app/src/main/java/com/svmdev/appmovies/data/model/Movie.java
package com.svmdev.appmovies.data.model;

import com.google.gson.annotations.SerializedName;
import com.svmdev.appmovies.data.webservice.URLs;

public class Movie {

    @SerializedName("id")
    private int id;

    @SerializedName("title")
    private String title;

    @SerializedName("poster_path")
    private String posterPath;

    @SerializedName("release_date")
    private String releaseDate;
    
    public String getTitle() {
        return title;
    }

    public String getPosterPath() {
        if(posterPath == null) {
            return "";
        }
        String replace = posterPath.replace("'\'","");
        return URLs.imageUrl + replace;
    }

    public String getPoster() {
        return posterPath;
    }

    public String getFormatedReleaseDate() {
        String[] formatedDate = releaseDate.split("-");
        return formatedDate[2]+"/"+formatedDate[1]+"/"+formatedDate[0];
    }

    public int getId() {
        return id;
    }

}

File: app/src/main/java/com/brunodiegom/movies/model/Movie.kt
package com.brunodiegom.movies.model

import androidx.recyclerview.widget.DiffUtil
import com.google.gson.annotations.SerializedName

/**
 * Data model returned from server.
 */
data class Movie(
    @SerializedName("id")
    val id: Int,
    @SerializedName("poster_path")
    val posterUrl: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("release_date")
    val releaseDate: String
) {
    companion object {
        const val THUMBNAIL_BASE_URL = "https://image.tmdb.org/t/p/w185"

        val DIFF_CALLBACK = object : DiffUtil.ItemCallback<Movie>() {
            override fun areItemsTheSame(oldItem: Movie, newItem: Movie): Boolean {
                return oldItem.id == newItem.id
            }

            override fun areContentsTheSame(oldItem: Movie, newItem: Movie): Boolean {
                return oldItem == newItem
            }
        }
    }
}
